lua/bufferline/buffer.lua

```diff
@@ -97,7 +97,7 @@ return {
     local no_name_title = options.no_name_title()
     local maximum_length = options.maximum_length()
 
-    if name then
+    if name and name ~= '' then
```

Comment:
It seems that this issue is caused by `nvim_buf_get_name(0)` returning `""` in for a buffer which is valid but unnamed— I expected it to be `nil`. So, we can simplify this a bit:

```lua
--- @type string
local name = buf_is_valid(buffer_number) and buf_get_name(buffer_number) or ''

local no_name_title = options.no_name_title()
local maximum_length = options.maximum_length()

if name ~= '' then
```

Now we only need to do one check on `name`, since it is guaranteed to be a `string`. If you could remove the other checks in this function which check for `name` being `nil` (e.g. line 113 `or not name`), that would be helpful too.

Reply:
Okay, I made a new commit with the changes. I hope I did it right :sweat_smile:

Do you want me to squash the commits?

Reply:
> Okay, I made a new commit with the changes. I hope I did it right 😅

Yeah, that's perfect!

> Do you want me to squash the commits?

Definitely. If you `git rebase -i HEAD~2` on your local machine, you should be able to `drop` the first commit entirely

Reply:
Right, just squashed them, now the branch has a single commit (: